### PR TITLE
Dungeon Tracker improvements

### PR DESCRIPTION
**Pull request by frankdj100**
- Added group pulse to keep dungeon run alive when at least one group member stays inside
- Added dungeon database table, updated with a lot of new quests
- Added speedcore support
- Multiple small improvements (see commits)
- Language override -- all dungeons only in English names

### CODE REVIEW

File: Hardcore.lua [hunk 1]
```diff
@@ -1630,65 +1632,164 @@ end
 -- DUNGEON RUN TRACKING --
 --------------------------
 
+-- dt_db ( = dungeon tracker database )
+-- 
+-- Contains all the info for the dungeons:
+-- { instanceID, zoneID, "English Name", type = { "D", "R", "B", "O" }, max_players, max_runs, { max_level_era, max_level_wotlk }, { quests } },
+-- Types: D = dungeon (5-player), R = raid, B = battleground, O = other
+
+local dt_db = {
```
Comment (aaronma37) [inline Hardcore.lua:1641]: might want to split this and last boss kills to another file somewhere

File: Hardcore.lua [hunk 2]
```diff
@@ -1630,65 +1632,164 @@ end
...
+	{ 450, "Hall of Legends", 1000, 1000, {1000, 1000}, {} },
+	--{ , "Borean Tundra", },
+	--{ , "Strand of the Ancients", },
+
+}
+
+local dt_db_id_to_name = nil
+local dt_db_max_levels = nil
+
+-- Hardcore:DungeonTrackerGetDungeonName( id )
```
Comment (aaronma37) [inline Hardcore.lua:1749]: rm

File: Hardcore.lua [hunk 3]
```diff
@@ -1993,6 +2101,97 @@ function Hardcore:DungeonTrackerCheckChanged( name )
...
+	if( CTL ) then
+		local name, serverName = UnitFullName("player")
+		local commMessage = COMM_COMMANDS[15] .. COMM_COMMAND_DELIM .. name .. COMM_FIELD_DELIM .. now .. COMM_FIELD_DELIM .. Hardcore_Character.dt.current.name
+		CTL:SendAddonMessage("NORMAL", COMM_NAME, commMessage, "PARTY")			-- Maybe to "INSTANCE_CHAT" instead?
+	end
+
+end
+
+
+-- DungeonTrackerTestReceivePulse()
```
Comment (aaronma37) [inline Hardcore.lua:2173]: rm

File: Hardcore.lua [hunk 4]
```diff
@@ -2001,8 +2200,8 @@ function Hardcore:DungeonTracker()
 	local name, instanceType, difficultyID, difficultyName, 
 		maxPlayers, dynamicDifficulty, isDynamic, instanceID, instanceGroupSize, LfgDungeonID = GetInstanceInfo()
 
---	local message = "Instance:" .. name .. ", " .. instanceType .. ", " .. instanceID 
---	Hardcore:Print( message )
+	--local message = "Instance:" .. name .. ", " .. instanceType .. ", " .. instanceID
```
Comment (aaronma37) [inline Hardcore.lua:2203]: rm

File: Hardcore.lua [hunk 5]
```diff
@@ -2086,9 +2304,11 @@ function Hardcore:DungeonTracker()
 		DUNGEON_RUN.date   		 = date("%m/%d/%y %H:%M:%S")
 		DUNGEON_RUN.time_inside  = 0
 		DUNGEON_RUN.time_outside = 0
-		DUNGEON_RUN.last_warn    = -1000
+		-- DUNGEON_RUN.last_warn    = -1000		-- Not necessary to initialise this; if it's nil, then it's -1000
```
Comment (aaronma37) [inline Hardcore.lua:2307]: rm

**Comment by aaronma37** [comment]
I think we should try to split out most of this code into files and put it into a rc

**Comment by frankdj100** [comment]
Replace with new PR in which dungeon code is moved to Dungeons.lua.

**Pull request closed without merging**